centreon/www/include/options/media/images/DB-Func.php

```diff
@@ -175,21 +175,34 @@ function deleteImg($imageId)
 
     $mediadir = "./img/media/";
 
-    $dbResult = $pearDB->query(
-        "SELECT dir_alias, img_path "
-        . "FROM view_img, view_img_dir, view_img_dir_relation "
-        . "WHERE img_id = $imageId AND img_id = img_img_id "
-        . "AND dir_dir_parent_id = dir_id"
-    );
-    while ($imagePath = $dbResult->fetch()) {
-        $fullpath = $mediadir . basename($imagePath["dir_alias"]) . "/" . basename($imagePath["img_path"]);
-        if (is_file($fullpath)) {
-            unlink($fullpath);
+        $query = "SELECT dir.dir_alias, img.img_path
+              FROM view_img img
+              INNER JOIN view_img_dir_relation rel ON img.img_id = rel.img_img_id
+              INNER JOIN view_img_dir dir ON rel.dir_dir_parent_id = dir.dir_id
+              WHERE img.img_id = :imageId";
+
+        $dbResult = $pearDB->prepare($query);
+        $dbResult->bindValue(':imageId', $imageId, PDO::PARAM_INT);
+        $dbResult->execute();
+        foreach ($dbResult->fetchAll(PDO::FETCH_ASSOC) as $imagePath) {
+            $fullpath = $mediadir . basename($imagePath["dir_alias"]) . "/" . basename($imagePath["img_path"]);
+            if (is_file($fullpath)) {
+                unlink($fullpath);
+            }
         }
-        $pearDB->query("DELETE FROM view_img WHERE img_id = $imageId");
-        $pearDB->query("DELETE FROM view_img_dir_relation WHERE img_img_id = $imageId");
+    try {
+        $pearDB->beginTransaction();
+        $deleteStatement = $pearDB->prepare('DELETE FROM view_img WHERE img_id = :imageId');
+        $deleteStatement->bindValue(':imageId', $imageId, PDO::PARAM_INT);
+        $deleteStatement->execute();
+
+        $deleteRelationStatement = $pearDB->prepare('DELETE FROM view_img_dir_relation WHERE img_img_id = :imageId');
+        $deleteRelationStatement->bindValue(':imageId', $imageId, PDO::PARAM_INT);
+        $deleteRelationStatement->execute();
+        $pearDB->commit();
+    } catch (PDOException $e) {
+        $pearDB->rollBack();
```

Comment:
As you catch exceptions, if an error occurs, the problem will be hidden.
In this case, throw the exception.
```suggestion
        $pearDB->rollBack();
        throw $e;
```